src/dialect.rs

```diff
@@ -2,6 +2,26 @@ use crate::allocator::{Allocator, NodePtr};
 use crate::cost::Cost;
 use crate::reduction::Response;
 
+// division with negative numbers are disallowed
+pub const NO_NEG_DIV: u32 = 1;
+
+// in conditions output, integers must use canonical encoding (i.e. no redundant
+// leading zeros)
+pub const COND_CANON_INTS: u32 = 2;
+
+// unknown condition codes are disallowed
```

Comment:
The scope is wrong here. `NO_UNKNOWN_CONDS` and `COND_CANON_INTS` are not properties of the dialect; they're used by `run_generator` which includes a dialect and other flags that affect post-processing. Can you move these somewhere which makes it clear it's not part of the dialect?

As an example, if we move `run_generator` to `chia_rs`, then these constants will have to move there anyway.

Reply:
yeah, I'm hoping we can do that too. I can move the definitions of these flags, but I don't want to prepare that transition too much before we have a better sense of what it will look like and what will work